### PR TITLE
Utility for printing JWT

### PR DESCRIPTION
**Pull request by bclarkx2**
### Description

In https://github.com/nicheinc/sfdcclient/pull/8, we would like to replace the deprecated [dgrijalva/jwt-go](https://github.com/dgrijalva/jwt-go) library with the maintained, drop-in replacement [golang-jwt/jwt](https://github.com/golang-jwt/jwt).

As part of the testing considerations for that PR, I would like to compare the JWTs produced using each JWT library. To make that easier, I added a small Go [cmd/print-jwt](./cmd/print-jwt) CLI tool to this repository. This tool accepts a series of flags dictating the Salesforce connection parameters you would like to use, performs a login to the Salesforce org specified using the `sfdcclient` package, then simply prints out the JWT used to the console.

I think this is a useful tool to have available for debugging and testing purposes more broadly, too.

This PR also removes the `Client` interface from this package that `NewClientWithJWTBearer` used to return  to follow our [Go client libraries](https://bookstack.niche.team/books/back-end-patterns-practices/page/inter-service-communication#bkmrk-go-client-libraries) best practices. This necessitates a major version release, as it is not technically fully backwards compatible to change the return type of this constructor.

### Testing Considerations

Run `go run ./cmd/print-jwt` and verify that the string that is printed to the console is a valid JWT prepped to be sent to Salesforce.

### Deployment

None -- non-functional service library update.

### Versioning

Major

### CODE REVIEW

File: jwtBearer.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type jwtBearer struct {
 	errMutex *sync.RWMutex
 }
 
-func NewClientWithJWTBearer(sandbox bool, instanceURL, consumerKey, username string, privateKey []byte, tokenDuration time.Duration, httpClient http.Client) (Client, error) {
+func NewClientWithJWTBearer(sandbox bool, instanceURL, consumerKey, username string, privateKey []byte, tokenDuration time.Duration, httpClient http.Client) (*jwtBearer, error) {
```
Comment (bclarkx2) [inline jwtBearer.go:42]: I changed this return type from an interface type (`Client`) to a struct type (`*jwtBearer`) to comply with [Go client libraries](https://bookstack.niche.team/books/back-end-patterns-practices/page/inter-service-communication#bkmrk-go-client-libraries).

I didn't think it was particularly valuable to also make sure that the returned struct type was exported, as I'm not sure what the use would be for other modules to be able to reference the jwtBearer _type_ directly. Let me know if you think I'm missing something and should convert this to an exported struct type!

File: jwtBearer.go [hunk 2]
```diff
@@ -78,17 +78,7 @@ func (c *jwtBearer) newAccessToken() error {
...
-		jwt.StandardClaims{
-			Issuer:    c.consumerKey,
-			Audience:  c.authServerURL,
-			Subject:   c.username,
-			ExpiresAt: time.Now().Add(c.tokenDuration).UTC().Unix(),
-		},
-	)
-	// Sign JWT with the private key
-	signedJWT, err := token.SignedString(c.rsaPrivateKey)
+	signedJWT, err := c.JWT()
```
Comment (kmurgic) [inline jwtBearer.go:81]: I like the way this refactor worked out! :rocket: Makes sense to make this its own separate method and call it here.

File: jwtBearer.go (hunk 1 above)
Comment (kmurgic) [reply jwtBearer.go:42]: It seems reasonable to me! I agree that ohter modules are unlikely to have a need to reference the jwtBearer type directly.

File: cmd/print-jwt/main.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	client, err := sfdcclient.NewClientWithJWTBearer(
+		sandbox,              // whether the instance the client connects to, is a sandbox or not
+		salesforceInstanceID, // the salesforce instance URL
+		consumerKey,          // the connected app consumer key
+		username,             // the username using the connected app
+		privateKeyBytes,
+		3*time.Second, // request timeout for the OAuth new token HTTP request (3 minute max)
+		http.Client{ // underlying HTTP client making all HTTP calls
```
Comment (kmurgic) [inline cmd/print-jwt/main.go:48]: Good comments. A couple of them felt redundant with the variable names on first look, but I think they all add a little bit information that is not in the variable name.

**Review by kmurgic — APPROVED** [review]
Nifty! Looks good to me!

**Comment by bclarkx2** [comment]
Tested via https://github.com/nicheinc/salesforce/pull/105

**Pull request merged**